source/common/router/config_impl.cc

```diff
@@ -44,6 +44,18 @@ RetryPolicyImpl::RetryPolicyImpl(const envoy::api::v2::RouteAction& config) {
   retry_on_ |= RetryStateImpl::parseRetryGrpcOn(config.retry_policy().retry_on());
 }
 
+CorsPolicyImpl::CorsPolicyImpl(const envoy::api::v2::CorsPolicy& config) {
+  for (const auto& origin : config.allow_origin()) {
+    allow_origin_.push_back(origin);
+  }
+  allow_methods_ = config.allow_methods();
+  allow_headers_ = config.allow_headers();
+  expose_headers_ = config.expose_headers();
+  max_age_ = config.max_age();
+  allow_credentials_ = PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, allow_credentials, false);
+  enabled_ = PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, enabled, true);
```

Comment:
Unless I'm missing something, isn't this going to make a fully empty CORS config be "enabled" which will kick on the filter? It seems like if the user does't specify any CORS config at all the default instantiation should be for it to be disabled?

Reply:
True! ... is there any way to detect absence of config? Maybe by making it an `unique_ptr` after all?
Setting it to false would mean that a user has to enable it on every route even if all the settings are contained on the vhost.

Reply:
You could make it a unique_ptr and return a const pointer. The issue though is when you translate v1 JSON config to v2. You will probably have to pass a pointer to a v2 object vs. reference, and then pivot on whether it is nullptr or not.